src/sentry/api/endpoints/organization_access_request_details.py

```diff
@@ -72,10 +72,11 @@ def get(self, request: Request, organization) -> Response:
                     team__organization=organization, member__user__is_active=True
                 ).select_related("team", "member__user")
             )
-        elif request.access.has_scope("team:write") and request.access.teams:
+        elif request.access.has_scope("team:write") and request.access.api_team_members:
             access_requests = list(
                 OrganizationAccessRequest.objects.filter(
-                    member__user__is_active=True, team__in=request.access.teams
```

Comment:
Unfortunately the `access.teams` paradigm,  along with `access.projects` and so on, are very useful but very indefensible in a silo split world.  Controllers have to be tamed about their data access.